instrumentation/jdbc/src/main/java/io/opentelemetry/javaagent/instrumentation/jdbc/JdbcTracer.java

```diff
@@ -69,19 +71,44 @@ public Span startSpan(PreparedStatement statement) {
   }
 
   public Span startSpan(Statement statement, String query) {
+    return startSpan(statement, normalizeAndExtractInfo(query));
+  }
+
+  public Span startSpan(Statement statement, SqlStatementInfo queryInfo) {
     Connection connection = connectionFromStatement(statement);
     if (connection == null) {
       return null;
     }
 
     DBInfo dbInfo = extractDbInfo(connection);
 
-    return startSpan(dbInfo, query);
+    return startSpan(dbInfo, queryInfo);
+  }
+
+  @Override
+  protected String normalizeQuery(SqlStatementInfo query) {
+    return query.getFullStatement();
   }
 
   @Override
-  protected String normalizeQuery(String query) {
-    return JDBCUtils.normalizeSql(query);
+  protected String spanName(DBInfo connection, SqlStatementInfo query, String normalizedQuery) {
+    String dbName = dbName(connection);
+    if (query.getOperation() == null && query.getTable() == null) {
```

Comment:
I think this makes more clear that query.getOperation() can't be null below
```suggestion
    if (query.getOperation() == null) {
```